### PR TITLE
Update torchpippy

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Work towards bringing torch 2.4's pippy implementation into Accelerate. Since `torchpippy` itself has not a ton of usage, we're just going to bring it in at once. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by SunMarc — APPROVED** [review]
Thanks for updating @muellerzr  ! If you agree with the suggestion below, can you also make the change in our docs ?

File: src/accelerate/inference.py [hunk 1]
```diff
@@ -155,10 +154,7 @@ def prepare_pippy(
             If `True`, the output from the last GPU (which holds the true outputs) is sent across to all GPUs.
     """
     if not is_pippy_available():
-        raise ImportError(
-            "`pippy` was not found to be installed on your system. Please "
-            "install using `pip install torchpippy` or ensure you have at least version 0.2.0"
-        )
+        raise ImportError("Using PiPPy requires PyTorch 2.4.0 or later.")
```
[the remark this answers is not in the document]
Comment (muellerzr) [reply src/accelerate/inference.py:157]: Yes agreed, slight oversight!

**Comment by muellerzr** [comment]
Converting to a draft for now as I'm facing a few issues with this. We know that `torchpippy` works for now

**Comment by muellerzr** [comment]
Update w/ this, all work bar T5

**Comment by muellerzr** [comment]
Finally merging, torch's integration doesn't support encoder/decoder tracing, so as a result I've raised an error stating to use torchpippy + a lower accelerate version (and as a result, we can't guarantee support)

**Pull request merged**